### PR TITLE
Restricted network installs are no longer UPI-only

### LINKED ISSUE #29541: [enterprise-4.6] Issue in file installing/installing_vsphere/installing-restricted-networks-vsphere.adoc
### Which section(s) is the issue in?

https://docs.openshift.com/container-platform/4.6/installing/installing_vsphere/installing-restricted-networks-vsphere.html#installation-about-restricted-networks_installing-restricted-networks-vsphere

### What needs fixing?

"You complete an installation in a restricted network on only infrastructure that you provision, not infrastructure that the installation program provisions, so your platform selection is limited."

Note a few lines below:
"Restricted network installations always use user-provisioned infrastructure. Because of the complexity of the configuration for user-provisioned installations, consider completing a standard user-provisioned infrastructure installation before you attempt a restricted network installation. Completing this test installation might make it easier to isolate and troubleshoot any issues that might arise during your installation in a restricted network."

per internal discussions, vSphere IPI should be an acceptable and supported configuration for disconnected OCP deployments

### PR DESCRIPTION
**Pull request by codyhoag**
Resolves #29541

### CODE REVIEW

**Comment by codyhoag** [comment]
@jianlinliu can you verify? Thanks!

**Comment by codyhoag** [comment]
Preview: https://deploy-preview-29552--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-restricted-networks-aws.html#installation-about-restricted-networks_installing-restricted-networks-aws

**Comment by jianlinliu** [comment]
LGTM.

**Comment by mburke5678** [comment]
@codyhoag LGTM. 
I wonder, not being familiar with the install process, there are sections in some restricted install assemblies that seem to be UPI-specific: _Completing installation on user-provisioned infrastructure_, _Creating the user-provisioned infrastructure_. Do we need to expressly state in these sections that IPI folks don't need to do these steps or do the steps need to be modified for IPI? 
Also, the issue states that _vSphere IPI_is supported for restricted. Do we need `ifdef` statements for vSphere only or is IPI supported for all platforms?

**Comment by codyhoag** [comment]
@mburke5678 your points are all valid, and currently a pain point we're trying to address. Historically, UPI was the only way, so restricted networks were understood to be UPI. Recently, a capability emerged that allowed for IPI installs to install into a restricted network, but the docs have not been updated to reflect that yet (OSP being the exception). The IPI procedures will require new assemblies (most cloud providers support IPI restricted now). This is currently in the queue to work on.

In summary, this is a known issue. But I do think a good short term goal would be to at least update the restricted network titles to explicitly state they're for UPI, now that there are two possibilities. I'll follow up with some edits for that once the freeze is lifted. Thanks!

**Review by vikram-redhat — APPROVED** [review]
lgtm for merge

**Comment by codyhoag** [comment]
With pre-GA approval exception, I'm merging this.
